### ISSUE TITLE
Accept `map()` object - or any iterable - in `int.from_bytes()`

### ISSUE THREAD

**Issue by Zac-HD** [pschanely/CrossHair#291]
found via [Hypothesis CI](https://github.com/HypothesisWorks/hypothesis/actions/runs/10234869198/job/28314927008?pr=4034#step:6:671) for https://github.com/HypothesisWorks/hypothesis/pull/4034, it looks like `int.from_bytes` should accept any iterable:

```python-traceback
  File ".../lib/python3.10/ipaddress.py", line 1195, in _ip_int_from_string
    return int.from_bytes(map(cls._parse_octet, octets), 'big')
  File ".../crosshair/libimpl/builtinslib.py", line 4627, in _int_from_bytes
    realize(len(b))
  File ".../crosshair/libimpl/builtinslib.py", line 4509, in _len
    raise TypeError(f"object of type '{name_of_type(type(ls))}' has no len()")
TypeError: object of type 'map' has no len()
```

**Comment by pschanely** [comment]
Yes! On it.

**Comment by pschanely** [comment]
Fixed in v0.0.66!
